corehq/blobs/tests/test_mixin.py

```diff
@@ -378,7 +381,74 @@ def open(self):
 
         def listdir(self):
             summaries = self.s3_bucket.objects.filter(Prefix=self.path + "/")
-            return [o.key for o in summaries]
+            try:
+                return [o.key for o in summaries]
+            except ClientError as err:
+                if not is_not_found(err):
+                    raise
+            return []
+
+
+class TestBlobMixinWithMigratingDbBeforeCopyToNew(TestBlobMixinWithS3Backend):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestBlobMixinWithMigratingDbBeforeCopyToNew, cls).setUpClass()
+        cls.db = PutInOldBlobDB(cls.db, TemporaryFilesystemBlobDB())
+
+    class TestBlob(TestBlobMixinWithS3Backend.TestBlob):
+
+        def __init__(self, db, name, bucket):
+            self.db = db.new_db
+            self.path = db.new_db.get_path(name, bucket)
+            self.fspath = db.old_db.get_path(name, bucket)
+
+        @property
+        def super(self):
```

Comment:
Heh, this works though, and avoids the subclass problem as long as you don't have two classes in your hierarchy with the same name:
```py
@property
def __super(self):
    return super(Class, self)
```